dnf5/commands/download/download.cpp

```diff
@@ -161,10 +181,27 @@ void DownloadCommand::run() {
     }
 
     if (url_option->get_value()) {
+        // If no urlprotocols are specified, all values within the urlprotocol_valid_options will be used
+        if (urlprotocol_option.empty()) {
+            urlprotocol_option = urlprotocol_valid_options;
+        }
         for (auto & [nerva, pkg] : download_pkgs) {
             auto urls = pkg.get_remote_locations();
             libdnf_assert(!urls.empty(), "Failed to get mirror for package: \"{}\"", pkg.get_name());
-            std::cout << urls[0] << std::endl;
+            auto valid_url = std::find_if(urls.begin(), urls.end(), [this](std::string url) {
+                for (auto protocol : urlprotocol_option) {
+                    // Ensure the urls matches the protocol specified by concating the colon.
+                    // i.e https: or http: not httpnextgen:
+                    if (url.starts_with(protocol + ":")) {
+                        return true;
+                    }
+                }
+                return false;
+            });
+            if (valid_url == urls.end()) {
+                libdnf_assert(true, "Failed to get mirror for package: \"{}\"", pkg.get_name());
```

Comment:
With the protocol filtering the asserts are probably too strong. I think it can easily happen that a specific package isn't available over selected protocol.

I would change it to logging:
```cpp
ctx.base.get_logger()->warning("Failed to get mirror for package: \"{}\"", pkg.get_name());
```

Reply:
This same solution was proposed for line 190 when I implemented the url option. Should I change it there as well for consistency? Or technically I can remove line 190 because I can catch it in the if condition